packages/Webkul/Installer/src/Console/Commands/DefaultUser.php

```diff
@@ -0,0 +1,455 @@
+            'APP_LOCALE',
+            'Please select the default timezone',
+            $this->getTimeZones()
+        );
+
+        $uiLocale = $this->option('ui_locale');
+
+        if ($uiLocale && ! in_array($uiLocale, $this->localCodes)) {
+            $this->warn("The specified locale code '$uiLocale' is not valid. Please select a valid locale.");
+            Log::warning("The specified locale code '$uiLocale' is not valid. Please select a valid locale.");
+
+            $uiLocale = null;
+        }
+
+        $defaultLocale = $uiLocale ?: $this->askForDefaultValues(
+            'APP_LOCALE',
+            'Please select the default application locale',
+            $this->locales
+        );
+
+        $localeId = DB::table('locales')->where('code', $defaultLocale)->where('status', 1)->first()?->id ?? 58;
+        $role = $isAdmin ? DB::table('roles')->where('permission_type', 'all')->first()?->id : DB::table('roles')->where('permission_type', 'custom')->first()?->id;
+
+        if (! $role) {
+            DB::table('roles')->updateOrInsert(
+                [
+                    'name'            => $isAdmin ? 'Admin' : 'User',
+                    'description'     => $isAdmin ? 'This role users will have all the access' : 'This role users have limited access',
+                    'permission_type' => $isAdmin ? 'all' : 'custom',
+                    'permissions'     => ! $isAdmin ? json_encode(['dashboard']) : null,
+                ]
+            );
+        }
+
+        $role = $isAdmin ? DB::table('roles')->where('permission_type', 'all')->first()?->id : DB::table('roles')->where('permission_type', 'custom')->first()?->id;
+
+        try {
+            DB::table('admins')->updateOrInsert(
+                [
+                    'api_token'    => Str::random(80),
+                    'created_at'   => date('Y-m-d H:i:s'),
+                    'name'         => $userName,
+                    'email'        => $userEmail,
+                    'password'     => $password,
+                    'role_id'      => $role,
+                    'status'       => 1,
+                    'timezone'     => $timezone,
+                    'ui_locale_id' => $localeId,
+                    'updated_at'   => date('Y-m-d H:i:s'),
+                ]
+            );
+
+            $this->info('-----------------------------');
+            $this->info('Congratulations! The User has been created successfully.');
+            $this->info('Please navigate to: '.env('APP_URL').'/admin'.' and use the following credentials for authentication:');
+            $this->info('Email: '.$userEmail);
+            $this->info('Password: '.$userPassword);
+            $this->info('Cheers!');
+            Log::info('Congratulations! The User has been created successfully');
+        } catch (\Exception $e) {
+            if (strpos($e->getMessage(), 'Duplicate entry')) {
+                $this->error('User with email '.$userEmail.' already exists.');
```

Comment:
How is this condition being used if the email is already being checked for uniqueness in the above code?

Reply:
resolved